pkg/graveler/graveler.go

```diff
@@ -1619,25 +1649,25 @@ func (g *Graveler) GetRangeIDByKey(ctx context.Context, repository *RepositoryRe
 }
 
 func (g *Graveler) Set(ctx context.Context, repository *RepositoryRecord, branchID BranchID, key Key, value Value, opts ...SetOptionsFunc) error {
-	isProtected, err := g.protectedBranchesManager.IsBlocked(ctx, repository, branchID, BranchProtectionBlockedAction_STAGING_WRITE)
-	if err != nil {
-		return err
-	}
-	if isProtected {
-		return ErrWriteToProtectedBranch
-	}
-
-	if repository.ReadOnly {
-		return ErrReadOnlyRepository
-	}
-
 	options := &SetOptions{}
 	for _, opt := range opts {
 		opt(options)
 	}
+	if !options.Force {
+		if repository.ReadOnly {
+			return ErrReadOnlyRepository
+		}
+		isProtected, err := g.protectedBranchesManager.IsBlocked(ctx, repository, branchID, BranchProtectionBlockedAction_STAGING_WRITE)
```

Comment:
This is a subtle but important change: we now allow overriding branch protection by using force.  So  ACL Developers and SuperUsers and various RBAC policies are no longer limited by not having `branches:SetBranchProtectionRules` (`SetBranchProtectionRulesAction`).  They can just force their way around branch protection!

Reply:
Also the fact that this change is debatable probably means that this business logic needs to live in a separate func: right now changing it means changing every instance. ;-(

Reply:
@nopcoder @guy-har WDYT?

Reply:
I think we have 3 options here:

1. Remain it has it's.
2. Change the `Force` flag to override only read-only repository protection.
3. Add a new required permission for users to use the `Force` flag.

Reply:
For now the force flag was to bypass the readonly flag - not bypass any access control. Bypass branch protection will require specific permissions, which can work in parallel to the force flag.

Reply:
IMO Also overriding readonly should require specific permissions!